tests/eosio.rentbw_tests.cpp

```diff
@@ -94,39 +94,138 @@ struct rentbw_tester : eosio_system_tester
+      if (net_fee > 0 && cpu_fee > 0) {
+         ilog("net_fee: ${net_fee} cpu_fee: ${cpu_fee}",("net_fee",net_fee)("cpu_fee",cpu_fee));
+         fee = asset(net_fee + cpu_fee, symbol{CORE_SYM});
+      }
+      return fee;
+   }
+
+   int64_t calc_rentbw_fee(const rentbw_state_resource& state, int64_t utilization_increase) {
+      if( utilization_increase <= 0 ) return 0;
+
+      // Let p(u) = price as a function of the utilization fraction u which is defined for u in [0.0, 1.0].
+      // Let f(u) = integral of the price function p(x) from x = 0.0 to x = u, again defined for u in [0.0, 1.0].
+
+      // In particular we choose f(u) = min_price * u + ((max_price - min_price) / exponent) * (u ^ exponent).
+      // And so p(u) = min_price + (max_price - min_price) * (u ^ (exponent - 1.0)).
+
+      // Returns f(double(end_utilization)/state.weight) - f(double(start_utilization)/state.weight) which is equivalent to
+      // the integral of p(x) from x = double(start_utilization)/state.weight to x = double(end_utilization)/state.weight.
+      // @pre 0 <= start_utilization <= end_utilization <= state.weight
+      auto price_integral_delta = [&state](int64_t start_utilization, int64_t end_utilization) -> double {
+         double coefficient = (state.max_price.get_amount() - state.min_price.get_amount()) / state.exponent;
+         double start_u     = double(start_utilization) / state.weight;
+         double end_u       = double(end_utilization) / state.weight;
+         return state.min_price.get_amount() * end_u - state.min_price.get_amount() * start_u +
+                  coefficient * std::pow(end_u, state.exponent) - coefficient * std::pow(start_u, state.exponent);
+      };
+
+      // Returns p(double(utilization)/state.weight).
+      // @pre 0 <= utilization <= state.weight
+      auto price_function = [&state](int64_t utilization) -> double {
+         double price = state.min_price.get_amount();
+         // state.exponent >= 1.0, therefore the exponent passed into std::pow is >= 0.0.
+         // Since the exponent passed into std::pow could be 0.0 and simultaneously so could double(utilization)/state.weight,
+         // the safest thing to do is handle that as a special case explicitly rather than relying on std::pow to return 1.0
+         // instead of triggering a domain error.
+         double new_exponent = state.exponent - 1.0;
+         if (new_exponent <= 0.0) {
+            return state.max_price.get_amount();
+         } else {
+            price += (state.max_price.get_amount() - state.min_price.get_amount()) * std::pow(double(utilization) / state.weight, new_exponent);
+         }
+
+         return price;
+      };
+
+      double  fee = 0.0;
+      int64_t start_utilization = state.utilization;
+      int64_t end_utilization   = start_utilization + utilization_increase;
+
+      if (start_utilization < state.adjusted_utilization) {
+         fee += price_function(state.adjusted_utilization) *
```

Comment:
it should be just an assignment, no?

Reply:
This code is essentially a copy-paste from `rentbw.cpp` so I left it as is.